pkg/controller/jobframework/reconciler.go

```diff
@@ -511,7 +511,13 @@ func (r *JobReconciler) ensureOneWorkload(ctx context.Context, job GenericJob, o
 		}
 
 		if _, finished := job.Finished(); !finished {
-			if err := r.stopJob(ctx, job, w, StopReasonNoMatchingWorkload, "No matching Workload"); err != nil {
+			var msg string
+			if w == nil {
+				msg = "Missing Workload, No restore PodSet info provided"
```

Comment:
Let's try to be more user friendly:
```suggestion
				msg = "Missing Workload; unable to restore pod templates"
```